server/utils/kvstore/kvstore.go

```diff
@@ -56,3 +59,46 @@ func StoreJSON(s KVStore, key string, v interface{}) (returnErr error) {
 	}
 	return s.Store(key, data)
 }
+
+func AtomicModify(s KVStore, key string, modify func(initialValue []byte) ([]byte, error)) error {
```

Comment:
Not sure why not including this as a method of the plugin store.

Reply:
It is a method of the store, which proxies to this function. This way of doing it allows us to synthetically put the method on the `KVStore interface`, rather than a struct. So any implementation of `KVStore` can "have" this function. This is particularly useful for the `hashedKeyStore`.